doc/examples/streamline_registration.py

```diff
@@ -0,0 +1,322 @@
+"""
+=======================
+Streamline Registration
+=======================
+
+This example shows how to register streamlines into a template space by
+applying non-rigid deformations.
+
+At times we will be interested in bringing a set of streamlines into some
+common, reference space to compute statistics out of the registered
+streamlines. For a discussion on the effects of spatial normalization
+approaches on tractography the work by Green et al. [Greene17]_ can be read.
+
+For brevity, we will include in this example only streamlines going through
+the corpus callosum connecting left to right superior frontal
+cortex. The process of tracking and finding these streamlines is fully
+demonstrated in the :ref:`streamline_tools` example. If this example has been
+run, we can read the streamlines from file. Otherwise, we'll run that example
+first, by importing it. This provides us with all of the variables that were
+created in that example.
+
+In order to get the deformation field, we will first use two b0 volumes. The
+first one will be the b0 from the Stanford HARDI dataset:
+
+"""
+
+import os.path as op
+
+if not op.exists('lr-superiorfrontal.trk'):
+    from streamline_tools import *
+else:
+    from dipy.data import read_stanford_hardi
+    hardi_img, gtab = read_stanford_hardi()
+    data = hardi_img.get_data()
+
+"""
+The second one will be the template FA map from the HCP1065 dataset:
+
+"""
+
+from dipy.data.fetcher import (fetch_mni_template, read_mni_template)
+
+fetch_mni_template()
+img_t2_mni = read_mni_template("a", contrast="T2")
+
+"""
+We filter the diffusion data from the Stanford HARDI dataset to find the b0
+images.
+
+"""
+
+import numpy as np
+
+b0_idx_stanford = np.where(gtab.b0s_mask)[0]
+b0_data_stanford = data[..., b0_idx_stanford]
+
+"""
+We then remove the skull from the b0's.
+
+"""
+
+from dipy.segment.mask import median_otsu
+
+b0_masked_stanford, _ = \
+    median_otsu(b0_data_stanford,
+                vol_idx=[i for i in range(b0_data_stanford.shape[-1])],
```

Comment:
I prefer `vol_idx=list(range(b0_data_stanford.shape[-1]))` because I think it is more clear

Reply:
Done in d614b88.